src/TestSuite/IntegrationTestCase.php

```diff
@@ -660,6 +661,33 @@ protected function _addTokens($url, $data)
         return $data;
     }
 
+    /**
+     * Recursively casts all data to string as that is how data would be POSTed in
+     * the real world
+     *
+     * @param array $data POST data
+     * @return array
+     */
+    protected function _castToString($data)
+    {
+        foreach ($data as $key => $value) {
+            if (is_scalar($value)) {
+                $data[$key] = (string)$value;
+
+                continue;
+            }
+
+            if (is_array($value)) {
+                $looksLikeFile = isset($value['error']) && isset($value['tmp_name']) && isset($value['size']);
+                if (!$looksLikeFile) {
```

Comment:
I am usually a fan of return early to keep nesting to a minimum. But no big deal here.

Reply:
In this case you need to continue the loop not return from the method. The loop is going to continue anyway after this check, so continuing early here won't have much benefit.

Reply:
Yes, thats why "no big deal" :)

Reply:
Already done :P